source/lac/trilinos_vector_base.cc

```diff
@@ -372,7 +372,7 @@ namespace TrilinosWrappers
     // is zero on _all_ processors.
     const Epetra_MpiComm *mpi_comm
       = dynamic_cast<const Epetra_MpiComm *>(&vector->Map().Comm());
-    Assert(mpi_comm != 0, ExcInternalError());
+    AssertThrow(mpi_comm != 0, ExcInternalError());
```

Comment:
I think we should at least test in a Debug build that the cast is successful. It should be safe to say that the type of that object does not change from Debug to Release. Hence I am now convinced that an `Assert` is the right thing to do.